civis/base.py

```diff
@@ -105,6 +111,9 @@ def _make_request(self, method, path=None, params=None, data=None,
             auth_error = response.headers["www-authenticate"]
             raise CivisAPIKeyError(auth_error) from CivisAPIError(response)
 
+        if check_retry_valid(method, response.status_code):
+            raise APIRetryError
```

Comment:
Oh, okay, I see what you are doing here.  You are raising a specific exception so that you can retry this whole function but match on the method and status code, but I think you could make this cleaner by just retrying around the `sess.request` part.

Take a look at the retry code we have in the SDK: https://github.com/civisanalytics/civis-integrations-sdk/blob/ddde0691232da937165553cb2ed738c11eeee4f8/civis_integration/utils/helpers/http_helpers.py#L65-L73

I think you can use that `retry_if_result` to pass in something like your `check_retry_valid` function.  The SDK logic is simpler because it is specifically for GET, but I bet you can make it work and then instead of raising these exceptions you can put the retries around just the request calls?